File: examples/clojure/yaml-bench.clj
#!/usr/bin/env clojure

;; Simple YAML benchmark using clj-yaml
;; Run with: clojure -Sdeps '{:deps {clj-commons/clj-yaml {:mvn/version "1.0.27"}}}' yaml-bench.clj <file.yaml>

(require '[clj-yaml.core :as yaml])

(defn -main [& args]
  (when (empty? args)
    (println "Usage: clojure -Sdeps '{:deps {clj-commons/clj-yaml {:mvn/version \"1.0.27\"}}}' yaml-bench.clj <yaml-file>")
    (System/exit 1))

  (let [filename (first args)

        ;; Read YAML file
        start-read (System/nanoTime)
        content (slurp filename)
        data (yaml/parse-string content)
        read-time (/ (- (System/nanoTime) start-read) 1e6)

        ;; Dump it back
        start-write (System/nanoTime)
        output (yaml/generate-string data)
        write-time (/ (- (System/nanoTime) start-write) 1e6)]

    (println output)
    (binding [*out* *err*]
      (println (format "\nRead:  %.2f ms" read-time))
      (println (format "Write: %.2f ms" write-time))
      (println (format "Total: %.2f ms" (+ read-time write-time))))))

(apply -main *command-line-args*)

File: examples/python/yaml-bench.py
#!/usr/bin/env python3
"""
Simple YAML benchmark using PyYAML
Usage: python3 yaml-bench.py <file.yaml>
"""

import sys
import time
import yaml

if len(sys.argv) < 2:
    print("Usage: python3 yaml-bench.py <yaml-file>")
    sys.exit(1)

filename = sys.argv[1]

# Read YAML
start_read = time.perf_counter()
with open(filename, 'r') as f:
    data = yaml.safe_load(f)
read_time = (time.perf_counter() - start_read) * 1000

# Write YAML
start_write = time.perf_counter()
output = yaml.dump(data, default_flow_style=False)
write_time = (time.perf_counter() - start_write) * 1000

print(output)
print(f"\nRead:  {read_time:.2f} ms", file=sys.stderr)
print(f"Write: {write_time:.2f} ms", file=sys.stderr)
print(f"Total: {read_time + write_time:.2f} ms", file=sys.stderr)
